File: SEP_T3/src/main/java/sep/java/sep_t3/contracts/IUserDao.java
package sep.java.sep_t3.contracts;

import com.fasterxml.jackson.core.JsonProcessingException;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import sep.java.sep_t3.model.User;

import java.util.List;

public interface IUserDao {
    public ResponseEntity<User> AddUser(String userAsJson) throws JsonProcessingException;
    public ResponseEntity<User> GetUser(String Email);
    public ResponseEntity<HttpStatus> DeleteUser(String email);
    public ResponseEntity<User> UpdateUser(String userAsJson) throws JsonProcessingException;
    public ResponseEntity<List<User>> GetAllUsers();
}

File: SEP3_T2/Contracts/DAO/IUserDao.cs
using Entities.Model;

namespace Contracts.DAO; 

public interface IUserDao {
    public Task<User> AddUser(User user);
    public Task<User> GetUser(Guid RUI);
    public Task<User> GetUser(string email);
    public Task<ICollection<User>> GetAllUsers();
    public Task DeleteUser(Guid RUI);
    public Task<User> UpdateUser(User user);
}
